packages/core/admin/admin/src/pages/Auth/AuthPage.tsx

```diff
@@ -37,17 +37,24 @@ const AuthPage = () => {
 
   const { token } = useAuth('AuthPage', (auth) => auth);
 
+  const searchParams = new URLSearchParams(search);
+  const redirectTo = searchParams.get('redirectTo');
+
   if (!authType || !forms) {
     return <Navigate to="/" />;
   }
 
   const Component = forms[authType as keyof FormDictionary];
 
+  if (token && redirectTo) {
+    return <Navigate to={redirectTo} />;
+  }
+
   // Redirect the user to the login page if
   // the endpoint does not exist or
   // there is already an admin user oo
   // the user is already logged in
-  if (!Component || (hasAdmin && authType === 'register-admin') || token) {
```

Comment:
Wouldn't all of this be fixed if we changed that condition to 

`
if (!Component || (hasAdmin && authType === 'register-admin' && token)) 
`

otherwise we leave it to each page to redirect to where it should redirect at the end of the process 🤔

Reply:
🤯